discogs_client/client.py

```diff
@@ -141,8 +141,21 @@ def search(self, *query, **fields):
         function are serialized into the request's query string.
         """
         if query:
-            fields['q'] = ' '.join(query)
-
+            unicode_query = []
+            for q in query:
+                try:
+                    unicode_q = q.decode('utf8')
+                except UnicodeDecodeError:
+                    # q is unicode.
+                    unicode_q = q
+                except UnicodeEncodeError:
+                    # q is a str.
+                    unicode_q = q
+                except AttributeError:
+                    # we are in python3.x and did not encode first.
+                    unicode_q = q
```

Comment:
We could do a catch-all. I made it `except Exception:` as stated in PEP 8, although I'm pretty sure Guido would still be shaking his head :). That should catch all exceptions we are looking for.